### PR TITLE
soc: xtensa: esp32s3: add support for SPIRAM

### PR DESCRIPTION
**Pull request by LucasTambor**
Add support for external PSRAM for esp32s3.
Test esp32s3 for cache coexistence. Update test documentation.
Add section to allocate memory of WiFi and NET stack in SPIRAM

### CODE REVIEW

**Review by uLipe — DISMISSED** [review]

**Review by sylvioalves — DISMISSED** [review]

File: boards/xtensa/esp32s3_devkitm/Kconfig.defconfig [hunk 1]
```diff
@@ -18,3 +18,6 @@ config HEAP_MEM_POOL_SIZE
 choice BT_HCI_BUS_TYPE
 	default BT_ESP32 if BT
 endchoice
+
+config ESP_SPIRAM
+	default y
```
Comment (marekmatej) [inline boards/xtensa/esp32s3_devkitm/Kconfig.defconfig:23]: I think SPIRAM should be enabled in the application code rather on board level.

**Comment by marekmatej** [comment]
Could we please wait until https://github.com/zephyrproject-rtos/zephyr/pull/58454 gets merged?

File: tests/boards/espressif_esp32/cache_coex/testcase.yaml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 tests:
   boards.esp32.cache_coex:
-    platform_allow: esp32 esp32s2_saola
+    platform_allow:
+      - esp32
+      - esp32s2_saola
+      - esp32s3_devkitm
```
Comment (mniestroj) [inline tests/boards/espressif_esp32/cache_coex/testcase.yaml:6]:
```suggestion
    platform_allow: >
      esp32
      esp32s2_saola
      esp32s3_devkitm
```
This probably needs to be a string, not a list.

File: boards/xtensa/esp32s3_devkitm/Kconfig.defconfig (hunk 1 above)
Comment (mniestroj) [reply boards/xtensa/esp32s3_devkitm/Kconfig.defconfig:23]: Would be great if if would be selected automatically based on devicetree node corresponding to SPIRAM.

**Review by marekmatej — DISMISSED** [review]

File: soc/xtensa/espressif_esp32/esp32s3/soc.c [hunk 3]
```diff
@@ -104,6 +111,31 @@ void IRAM_ATTR __esp_platform_start(void)
 	/* Apply SoC patches */
 	esp_errata();
 
+
```
Comment (marekmatej) [inline soc/xtensa/espressif_esp32/esp32s3/soc.c:114]: unnecessary new line :)

File: tests/boards/espressif_esp32/cache_coex/README.rst [hunk 4]
```diff
@@ -12,32 +12,44 @@ with a random generated pattern. At the same time, a whole SPI Flash page is upd
 value. By the end of the thread iterations, both PSRAM and SPI Flash have its contents compared against
 expected values to check for integrity.
 
+Supported Boards
+****************
+- esp32
```
Comment (marekmatej) [inline tests/boards/espressif_esp32/cache_coex/README.rst:17]: please update the target esp32 -> esp32_devkitc_wroom ^ esp32_devkitc_wrover

File: tests/boards/espressif_esp32/cache_coex/README.rst [hunk 5]
```diff
@@ -12,32 +12,44 @@ with a random generated pattern. At the same time, a whole SPI Flash page is upd
...
 ********************
 
-Make sure you have the ESP32 DevKitC connected over USB port.
+Make sure you have the target connected over USB port.
 
 .. code-block:: console
 
-   west build -b esp32_devkitc_wrover tests/boards/espressif_esp32/cache_coex
-   west flash --esp-device /dev/ttyUSB0
+   west build -b <board> tests/boards/espressif_esp32/cache_coex
```
Comment (marekmatej) [inline tests/boards/espressif_esp32/cache_coex/README.rst:28]: good idea :)

**Review by marekmatej — APPROVED** [review]

File: soc/xtensa/espressif_esp32/esp32s3/default.ld [hunk 6]
```diff
@@ -233,9 +233,20 @@ SECTIONS
   {
     _ext_ram_bss_start = ABSOLUTE(.);
     *(.ext_ram.bss*)
-    . = ALIGN(0x10000);
+    . = ALIGN(4);
```
Comment (mniestroj) [inline soc/xtensa/espressif_esp32/esp32s3/default.ld]: This alignment was previously wrong / just arbitrary?

Comment (LucasTambor) [reply soc/xtensa/espressif_esp32/esp32s3/default.ld]: It was arbitrary, for testing purposes. Ended up forgetting to put it on the first commit. Thanks for pointing it out

**Review by mniestroj — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**